demos/bluetooth/mesh/thingy52/thingy52_srv/src/model_handler.c

```diff
@@ -0,0 +1,348 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-BSD-5-Clause-Nordic
+ */
+
+#include <bluetooth/bluetooth.h>
+#include <bluetooth/mesh/models.h>
+#include <drivers/gpio.h>
+#include <drivers/gpio/gpio_sx1509b.h>
+#include <drivers/pwm.h>
+#include "model_handler.h"
+#include "orientation_handler.h"
+#include "thingy52_srv.h"
+
+#define SPEAKER_PWM_FREQ 880
+#define RGB_WORK_ITEM_MAX 5
+
+void rgb_set_handler(struct bt_mesh_thingy52_srv *srv,
+		     struct bt_mesh_msg_ctx *ctx,
+		     struct bt_mesh_thingy52_rgb_msg rgb);
+
+struct devices {
+	struct device *io_expander;
+	struct device *gpio_0;
+	struct device *spkr_pwm;
+	struct device *orientation;
+};
```

Comment:
In master, `struct device` is now always `const`, so you'll have to fix that in the next rebase